src/ui/components/filters/sortoptionscomponent.js

```diff
@@ -23,31 +23,57 @@ export default class SortOptionsComponent extends Component {
     this.selectedOptionIndex = parseInt(this.core.globalStorage.getState(this.name)) || 0;
     this.options[this.selectedOptionIndex].isSelected = true;
     this.hideExcessOptions = this._config.showMore && this.selectedOptionIndex < this._config.showMoreLimit;
-    this.showReset = this._config.showReset && this.selectedOptionIndex !== 0;
+    this.searchOnChangeIsEnabled = this._config.searchOnChange;
+    this.showResetIsEnabled = this._config.showReset;
+    this.showReset = this.showResetIsEnabled && this.selectedOptionIndex !== 0;
```

Comment:
would it make sense to put these statelike things (showReset, isSearchResults) into the setState, and reference them with this.getState('showReset')?

Reply:
the issue with that is that it would trigger a re-render when we don't want it to

Reply:
oh, I was thinking that every time showReset or isSearchResults change we do a setState anyway (I think?). And if that's the case, you could just add them as properties to the state instead. Then any time you want to guard a setState you could reference `this.getState('showReset')` or `this.getState('isSearchResults')`

(`this.getState('showReset')` won't trigger a re-render but I don't think that's what you meant)

Reply:
We don't always want to call `setState()` every time `showReset` changes because then the component would re-render and the focus would be lost as the user navigates between the default and the second non-default option with the keyboard. It could work for changes to isNoResults, but I don't think it's worth refactoring just for that.